### PR TITLE
Update floor, ceil and round in Dates to be consistent with Base

### PR DESCRIPTION
**Pull request by ssikdar1**
PR for issue #34782 to have `floor`, `ceil`, and `round` in the `Dates` library to be consistent with Base where
`floor([T,] x)` converts the result to type T.

PR changes the positional arguments, update the docstrings and updates the `rounding.jl` unit tests.

## Testing
Tests pass with `$ ./julia stdlib/Dates/test/rounding.jl `

### CODE REVIEW

**Comment by rapus95** [comment]
Imo we rather should add functions to support `round(Date, now(), Month)` as that perfectly mirrors the `round(T, val, precision)` function. I.e. when writing `round(now(), Day)` and claming that it should be `round(Day, now())` we should rather define `round(Date, now()) == convert(Date, round(now(), Day))` where in the latter statement the rounding is unneccessary though.

Tl;Dr: Proposal:
Add `round([::Type{<:TimeType},] val::TimeType [, precision::Period [, ::RoundingMode]])`

**Comment by ssikdar1** [comment]
Added 
```
floor(dt::TimeType, p::Period) -> TimeType
ceil(dt::TimeType, p::Period) -> TimeType
round(dt::TimeType, p::Period, [r::RoundingMode]) -> TimeType
```
back in based on the comment in #34782. (https://github.com/JuliaLang/julia/issues/34782#issuecomment-588152602)

For functions like 
```
ceil(x::Period, precision::T) where T <: Union{TimePeriod, Week, Day} -> T
ceil(precision::T, x::Period) where T <: Union{TimePeriod, Week, Day} -> T
```
Not sure this can work since both Period and precision are of type `ConvertiblePeriod`.
( When I try to add them I get errors in the tests. )

Not sure about the proposal and convert?
